cluster/state_updator.go

```diff
@@ -183,9 +183,22 @@ func (s *StateUpdator) extractSequenceFromKey(key string) (int, error) {
 	return math.MaxInt64 - i, nil
 }
 
-// Update updates the state based on the previous state. The update function provides the operation to update the state
-// based on the previous state, returning the new state which will be stored. The function returns when the new state
-// has been committed to object storage, or an error occurs.
+func (s *StateUpdator) GetState() ([]byte, error) {
+	s.lock.Lock()
+	defer s.lock.Unlock()
+	return s.state, nil
+}
+
```

Comment:
I think it reads nicer when public methods are at top, or at the very least colocated (especially when they lack complex logic via private methods like Update does)